engine/runtime/src/main/scala/org/enso/compiler/context/SuggestionBuilder.scala

```diff
@@ -636,14 +637,21 @@ final class SuggestionBuilder[A: IndexedSource](
     * @param arg the value argument
     * @return the suggestion argument
     */
-  private def buildArgument(arg: IR.DefinitionArgument): Suggestion.Argument =
+  private def buildArgument(arg: IR.DefinitionArgument): Suggestion.Argument = {
+    val signatureOpt = arg.name.getMetadata(TypeSignatures).map { meta =>
+      buildTypeSignature(meta.signature) match {
+        case Vector(targ) => buildTypeArgumentName(targ)
```

Comment:
This seems to assume that you don't deal with `Function`, which can return vector of size > 1?